emulator/src/loader/program.rs

```
@@ -285,8 +296,8 @@ impl Program {
     pub fn write_mem(&mut self, address: u32, value: u32) -> Result<(), ExecutionResult> {
         let step = self.step;
         let section = self.find_section_mut(address)?;
-        if section.is_code {
-            return Err(ExecutionResult::WriteToCodeSection);
+        if !section.is_write {
+            return Err(ExecutionResult::WriteToReadOnlySection);
```

We need to raise an exception in case of an attempt to write a code section too. Because, someone could generate a program with a writable code section, and even though it could be a valid riscv program is not valid for us because if not we would need to support a memory challenge for the code section (that we challenge more easily assuming that is fixed)